enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,509 @@
+valuable given the reduction of CPU by ~21% and memory by ~33% when comparing
+the `minimal` to the `full` profile. More details can be consulted in this
+[document](https://docs.google.com/document/d/1MA-HTJQ_X7y_bwpJS2IPbGmC4qDyIMp25jisr34X2F4/edit?usp=sharing).
+
+Moreover, through Telemetry, we collect for each cluster the top 3 Prometheus
+jobs that generate the biggest amount of samples. With this data, we know that
+for recent OpenShift versions, the 5 components most often reported as the
+biggest producers are: the Kubernetes API servers, the Kubernetes schedulers,
+kube-state-metrics, kubelet and the network daemon.
+
+### User Stories
+
+- As a user, I want to lower the amount of resources consumed by Prometheus in a
+  supported way, so I can configure the clusters metrics collection profiles to
+  `minimal`.
+- As a developer, I want a supported way to collect a subset of the metrics
+  exported by my operator and operands, while still collecting necessary metrics
+  for alerts, visualization of key indicators and Telemetry.
+- As a developer of a component (that does not yet implement a profile), I want
+  to extract metrics needed to implement said profile, based on the assets I
+  provide, or the ones gathered from the cluster based on a group of target
+  selectors, and a plug-in relabel configuration to apply within the monitor.
+- As a developer of a component (that does not, or only partially implements a
+  profile), I want to get information about any monitors that are not yet
+  implemented for any of the supported profiles that are offered.
+- As a developer of a component (that implements a profile), I want to verify if
+  all the profile metrics are present in the cluster, and which of the profile
+  monitors are affected if not. Also, I want additional information to narrow
+  down where these metrics are exactly being used.
+
+### Goals
+
+- Give users a supported and documented method to pick the amount of metrics
+  being collected by the platform, in a way that allows
+  cluster-monitoring-operator and other operators to provide their own
+  defaulting.
+- Other components of the cluster (alerts, console, HPA and VPA and telemetry)
+  should not be negatively affected by this feature.
+- Developers may optionally adhere to this feature, those that do not adhere
+  should not be impacted.
+
+### Non-Goals
+
+- Dynamically adjusting metrics scraped at runtime based on heuristics.
+- Allowing users to explicitly set a list or a regex of metrics that they would
+want to be included in a given profile.
+- Support the addition of profiles other than `full` and `minimal`.
+
+## Proposal
+
+This enhancement leverages label selectors to allow cluster-monitoring-operator
+to select the appropriate monitors for a given metrics collection profile. 
+
+Given this, the following steps would be necessary to implement this
+enhancement:
+
+- expose a configuration option in the cluster-monitoring-operator ConfigMap
```

Comment:
Given we are moving away from the configmap to something more structured, what impact does adding new configmap items have on that effort? It can only hinder the effort no?

Reply:
It's another field to move but it already exists in the configmap being tech-preview.

Reply:
Creating more fields to move sounds counterproductive, is there no way we could just introduce this as structured from the first instance?

Reply:
@JoelSpeed I'm not sure to understand your point: the field already exists in the [CMO configmap](https://github.com/openshift/cluster-monitoring-operator/blob/23a896aec95f0155771e808d20d4abb84d978094/pkg/manifests/types.go#L239-L245) so it needs a migration anyway.